### ISSUE TITLE
cannot open source file "libavutil/avconfig.h"

### ISSUE THREAD

**Issue by ghost1372** [telegramdesktop/tdesktop#4397]
hi i want to build tdesktop but i get error

Error (active)	E1696	cannot open source file "libavutil/avconfig.h"

windows 10 / VS2017

![untitled](https://user-images.githubusercontent.com/9213496/35990210-98228d26-0d18-11e8-8ec2-47b5a9394eac.png)
![untitled2](https://user-images.githubusercontent.com/9213496/35990216-9ddbc91c-0d18-11e8-9c7a-23658135dc17.png)

**Comment by stek29** [comment]
You need MSYS2 bash, not WSL one

**Comment by ghost1372** [comment]
@stek29  tnx i fixed problem but now i have this one problem.
i cant build Telegram.sln
in Release Build:
![screenshot 1](https://user-images.githubusercontent.com/9213496/36022491-8fd012e8-0d9e-11e8-8aec-91afa5db3db7.png)
in Debug Build:
![screenshot 2](https://user-images.githubusercontent.com/9213496/36022505-996150ba-0d9e-11e8-8ff3-123239ccdb5d.png)

**Comment by stek29** [comment]
Never used Visual Studio, sorry

**Comment by willowTank** [comment]
How did you fix the problem?

**Comment by ghost1372** [comment]
@willowTank  Just make sure That all the prerequisites are properly installed and configurable 
Run all the commands in sequence

**Comment by willowTank** [comment]
![image](https://user-images.githubusercontent.com/24641385/59837779-1cb70780-934e-11e9-97a3-14dba2eaaa83.png)
hi Thank you.
I couldn't run the command in x86 Native Tools Command Prompt For VS 2019.
So I am trying to do in MSYS2.
But I am getting another error.
![image](https://user-images.githubusercontent.com/24641385/59837889-538d1d80-934e-11e9-9343-aabe7a334331.png)
For that problem, I am getting error in VS 2019.
![image](https://user-images.githubusercontent.com/24641385/59837926-60117600-934e-11e9-8470-773ef9c71674.png)
Best Regards

**Comment by ghost1372** [comment]
@willowTank  This error is due to the fact that some libraries have not been properly compiled And the reason for not compiling is mostly because of the lack of proper installation of the tools.
 The only thing I can remember is that I deleted all the items (source code, installed programs, environment variables), And I started everything again. just be sure to install the programs correctly and  add environment variables (Some programs do this automatically, just make sure to tick this option)

**And very important point:**
Some libraries have been added as refrence and there are no clones of code in the project
So after checking out the source code, check that these folders and their content are available and if not, download and copy yourself.
![Untitled](https://user-images.githubusercontent.com/9213496/59839234-927a9d80-9365-11e9-91d3-9c47478fa8b8.png)
